### PR TITLE
REPO:-1089: Adds fragment caching per tenant

### PR DESCRIPTION
**Pull request by okeen**

### CODE REVIEW

File: app/views/hyku/api/v1/work/_work.json.jbuilder [hunk 1]
```diff
@@ -1,122 +1,122 @@
 # frozen_string_literal: true
 # FIXME: many attributes here left nil so specs will pass
-json.uuid work.id
+json.cache! [@account.tenant, work.id, work.date_modified, work.member_of_collection_ids & collection_docs.pluck('id')] do
```
Comment (cjcolvar) [inline app/views/hyku/api/v1/work/_work.json.jbuilder:3]: See https://github.com/ubiquitypress/hyku-api/pull/30#discussion_r632717720

File: lib/hyku_addons/engine.rb [hunk 2]
```diff
@@ -92,6 +94,18 @@ def reload_hyrax_config!
...
+        def setup_tenant_cache
+          Rails.application.config.cache_store = :redis_cache_store, { url: Redis.current.id }
+          Rails.cache = ActiveSupport::Cache.lookup_store(Rails.application.config.cache_store)
+          ActionController::Base.perform_caching = Settings.enable_caching || true
+        end
+
+        def disable_tenant_cache
+          Rails.application.config.action_controller.perform_caching = false
+          Rails.application.config.cache_store = :memory_store
+          ActionController::Base.perform_caching = false
```
Comment (cjcolvar) [inline lib/hyku_addons/engine.rb:107]: Should this be `Settings.enable_caching || false`?

File: lib/hyku_addons/engine.rb [hunk 3]
```diff
@@ -92,6 +94,18 @@ def reload_hyrax_config!
             config.geonames_username = Settings.geonames_username
           end
         end
+
+        def setup_tenant_cache
+          Rails.application.config.cache_store = :redis_cache_store, { url: Redis.current.id }
+          Rails.cache = ActiveSupport::Cache.lookup_store(Rails.application.config.cache_store)
+          ActionController::Base.perform_caching = Settings.enable_caching || true
```
Comment (cjcolvar) [inline lib/hyku_addons/engine.rb:101]:
```suggestion
          ActionController::Base.perform_caching = Settings.enable_caching && true
```

**Review by cjcolvar — COMMENTED** [review]
Looks good just a couple suggestions.

File: lib/hyku_addons/engine.rb [hunk 4]
```diff
@@ -92,6 +94,20 @@ def reload_hyrax_config!
             config.geonames_username = Settings.geonames_username
           end
         end
+
+        def setup_tenant_cache
+          Rails.application.config.action_controller.perform_caching = Settings.cache_enabled && true
+          ActionController::Base.perform_caching = Rails.application.config.action_controller.perform_caching
+          Rails.application.config.cache_store = :redis_cache_store, { url: Redis.current.id }
+          Rails.cache = ActiveSupport::Cache.lookup_store(Rails.application.config.cache_store)
```
Comment (cjcolvar) [inline lib/hyku_addons/engine.rb:102]: Can these lines be skipped if `perform_caching == false`?

Comment (okeen) [reply lib/hyku_addons/engine.rb:102]: good point

**Review by cjcolvar — COMMENTED** [review]
Looks good.  I just had one small optimization question.
I'd like to hold off on merging this until the file deletion bugfix can make its way to main and get deployed.

**Comment by BertZZ** [comment]
Code moved to #446

**Pull request closed without merging**